File: frontend/src/utils/coimbatore.js
export const COIMBATORE_BOUNDS = {
  minLat: 10.95,
  maxLat: 11.15,
  minLng: 76.85,
  maxLng: 77.05,
};

export const COIMBATORE_CENTER = { lat: 11.0168, lng: 76.9558 };

export function isInCoimbatore(lat, lng) {
  return (
    lat >= COIMBATORE_BOUNDS.minLat &&
    lat <= COIMBATORE_BOUNDS.maxLat &&
    lng >= COIMBATORE_BOUNDS.minLng &&
    lng <= COIMBATORE_BOUNDS.maxLng
  );
}

export function validateCoimbatore(lat, lng, label) {
  if (!isInCoimbatore(lat, lng)) {
    throw new Error(`${label} is outside Coimbatore. Service is currently available only in Coimbatore.`);
  }
}

File: backend/app/core/coimbatore.py
"""
Coimbatore city bounding box and validation utilities.
"""

from fastapi import HTTPException

COIMBATORE_BOUNDS = {
    "min_lat": 10.95,
    "max_lat": 11.15,
    "min_lng": 76.85,
    "max_lng": 77.05,
}

COIMBATORE_CENTER = {"lat": 11.0168, "lng": 76.9558}

def is_in_coimbatore(lat: float, lng: float) -> bool:
    return (
        COIMBATORE_BOUNDS["min_lat"] <= lat <= COIMBATORE_BOUNDS["max_lat"]
        and COIMBATORE_BOUNDS["min_lng"] <= lng <= COIMBATORE_BOUNDS["max_lng"]
    )

def validate_coimbatore_location(lat: float, lng: float, location_name: str = "Location"):
    if not is_in_coimbatore(lat, lng):
        raise HTTPException(
            status_code=400,
            detail=f"{location_name} is outside Coimbatore. Service is currently available only in Coimbatore."
        )
